src/backend/api/Fusion.Resources.Domain/Queries/Departments/GetRelevantDeparmentProfile.cs

```diff
@@ -0,0 +1,322 @@
+
+                    }
+                    else
+                    {
+
+                        Reasons.Add("NOT Manager");
+
+                    }
+
+                    if (Reasons.Count > 0)
+                    {
+                        QueryRelevantDepartmentProfile? DepartmentProfile = await GetDepartmentInformation(department.Key.Trim('*').TrimEnd(), Reasons, request.Query);
+                        await StoreResult(lstDepartments, DepartmentProfile);
+                    }
+
+                }
+
+                return lstDepartments;
+            }
+
+
+            private async Task<bool> GetDepartmentChildren(string dep, List<string> reason, ODataQueryParams query)
+            {
+                if (sapIdFound == true)
+                {
+                    return true;
+                }
+                var DepChildren = await ResolveChildren(dep);
+                if (DepChildren != null)
+                {
+                    // recurse over all children categories and add them to the list
+                    foreach (var child in DepChildren.Children)
+                    {
+                        if (sapIdFound == true)
+                        {
+                            break;
+                        }
+                        QueryRelevantDepartmentProfile? ChildDepartmentProfile = await GetDepartmentInformation(child.DepartmentId, reason, query);
+                        await StoreResult(lstDepartments, ChildDepartmentProfile);
+
+                        await GetDepartmentChildren(child.DepartmentId, reason, query);
+                    }
+                }
+                return true;
+            }
+
+            private static async Task<bool> StoreResult(List<QueryRelevantDepartmentProfile?> lstDepartments, QueryRelevantDepartmentProfile? DepartmentProfile)
+            {
+                if (DepartmentProfile != null)
+                {
+                    lstDepartments.Add(DepartmentProfile);
+                }
+
+                return true;
+            }
+
+            private async Task<QueryRelevantDepartmentProfile?> GetDepartmentInformation(string departmentpath, List<string> Reasons, ODataQueryParams query)
+            {
+                var cacheKey = $"{departmentpath.Replace(" ", "")}";
+                QueryRelevantDepartmentProfile? DepartmentProfile = null;
+                ApiOrgUnit? departmentInfo;
+
+
+                var incache = memCache.TryGetValue(cacheKey, out departmentInfo);
+
+                if (incache == true)
+                {
+                    Console.WriteLine($"======================== FOUND CACHED {departmentInfo?.FullDepartment} ==========================================");
+
+                }
+                else
+                {
+                    departmentInfo = await lineOrgResolver.ResolveOrgUnitAsync(DepartmentId.FromFullPath(departmentpath));
+
+                    var cacheEntryOptions = new MemoryCacheEntryOptions()
```

Comment:
You should'nt provide cache options within query-code.
If required, should be done in constructor.

Usually during api-setup.
See if you can find other samples (not including CommonLib API)

Reply:
noted